test/v8-types-generated.test.js

```diff
@@ -0,0 +1,180 @@
+// Tap may intermittently fail with "no plan" or, if plans are added,
+// mistaken "incorrect number of tests" errors if run time is > ~25s.
+
+// We only have input from our own module, but still, the strings are complex, check they can't contain surprises
+test('Ensure eval sanitising works as expected before using fixture', function (t) {
+  t.throws(function () {
+    evalSafeString("'some string'; console.log('>>> Do bad things');")
+  }, { actual: 2, expected: 1 })
+
+  t.throws(function () {
+    evalSafeString("'some string' + console.log('>>> Do bad things');")
+  }, { actual: 'BinaryExpression', expected: 'Literal' })
+
+  t.throws(function () {
+    evalSafeString("`some string${ console.log('>>> Do bad things') }`")
+  }, { actual: 'TemplateLiteral', expected: 'Literal' })
+
+  // Simulate redefining 'RegExp' to be malicious when it is called later by an innocent 'new RegExp()'
+  t.throws(function () {
+    evalSafeString("function RegExp () { console.log('>>> Do bad things') }")
+  }, { actual: 'FunctionDeclaration' , expected: 'ExpressionStatement' })
+
+  t.throws(function () {
+    evalSafeString("RegExp = function () { console.log('>>> Do bad things') }")
+  }, { actual: 'AssignmentExpression' , expected: 'Literal' })
+
+  t.throws(function () {
+    evalSafeString("`some string${ function RegExp () { console.log('>>> Do bad things') } }`")
+  }, { actual: 'TemplateLiteral' , expected: 'Literal' })
+
+  t.throws(function () {
+    evalSafeString("'/abc/' && (RegExp = function () { console.log('>>> Do bad things') })")
+  }, { actual: 'LogicalExpression' , expected: 'Literal' })
+
+  t.throws(function () {
+    evalSafeString("(RegExp = function () { console.log('>>> Do bad things') }) && '/abc/'")
+    global.eval(str + "; new RegExp('/abc/');")
+  }, { actual: 'LogicalExpression' , expected: 'Literal' })
+
+  // Test regex creation
+  t.throws(function () {
+    evalSafeRegexDef("'/abc'; console.log('>>> Do bad things');")
+  }, { actual: 2 , expected: 1 })
+
+  t.throws(function () {
+    evalSafeRegexDef("'/abc' + console.log('>>> Do bad things')")
+  }, { actual: 'BinaryExpression', expected: 'Literal' })
+
+  t.throws(function () {
+    evalSafeRegexDef("(RegExp = function () { console.log('>>> Do bad things') }) && '/abc/'")
+  }, { actual: 'LogicalExpression' , expected: 'Literal' })
+
+  t.throws(function () {
+    evalSafeRegexDef("'/abc/' && (RegExp = function () { console.log('>>> Do bad things') })")
+  }, { actual: 'LogicalExpression' , expected: 'Literal' })
+
+  t.throws(function () {
+    evalSafeRegexDef("`/abc${ console.log('>>> Do bad things') }/`")
+  }, { actual: 'TemplateLiteral' , expected: 'Literal' })
+
+  t.throws(function () {
+    evalSafeRegexDef("`/abc${ function RegExp () { console.log('>>> Do bad things') } }/`")
+  }, { actual: 'TemplateLiteral' , expected: 'Literal' })
+
+  t.end()
+})
+
+test('Generate profile and test its output', function (outerTest) {
```

Comment:
I did try that, but I ran into a whole lot of strange issues where Tap was failing intermittently (possibly, depending on how long the test took to run; but the tests were _not_ timing out), with errors caused by it for some reason miscounting the number of expected tests (adding `t.plan(n)` didn't solve it, it just changed the intermittently failure to be Tap miscounting the tests run, instead of miscounting the tests expected). 

I'm not entirely sure why, but adding the outer and inner tests and the .then() chain was about the only thing that reliably prevented these (or at least, prevented them so long as the total test file duration wasn't above around 25 seconds, which should be okay since on an old macbook air they take 10 seconds).

Reply:
Thinking about it, these errors were when I had both sets of tests in one file. It was probably just Tap getting confused on timing out while there were still async tests in the pipe.

I've tried it again and it seems pretty reliable now (not tried it on Windows yet...)